File: apps/commerce-api/src/main/java/com/loopers/domain/product/ProductEventPublisher.java
package com.loopers.domain.product;

import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Component
public class ProductEventPublisher {
    private final ApplicationEventPublisher applicationEventPublisher;

    public void publishProductViewed(Long productId) {
        ProductViewedEvent event = new ProductViewedEvent(productId);
        applicationEventPublisher.publishEvent(event);
    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/domain/product/ProductEventPublisher.kt
package com.loopers.domain.product

import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component

@Component
class ProductEventPublisher(
    private val publisher: ApplicationEventPublisher
) {
    fun publishSoldOutEvent(productIds: List<Long>) {
        publisher.publishEvent(ProductSoldOutEvent(productIds))
    }
}
